src/test/java/it/gov/pagopa/payhub/activities/activity/SendEmailIngestionFlowActivityTest.java

```diff
@@ -0,0 +1,194 @@
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertTrue(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, true));
+    }
+
+    /**
+     * send email for KO loading flow, success: valid user and valid flow
+     */
+    @Test
+    void sendEmailIngestionKoValidUserValidFlowSuccess() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(validIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertTrue(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, false));
+    }
+
+
+    /**
+     * send email for OK loading flow, success: valid user and invalid flow
+     */
+    @Test
+    void sendEmailIngestionOkValidUserInvalidFlowFailed() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(invalidIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(invalidIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertFalse(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, true));
+    }
+
+    /**
+     * send email for KO loading flow, failed: valid user and invalid flow
+     */
+    @Test
+    void sendEmailIngestionKoValidUserInvalidFlowFailed() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(invalidIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(invalidIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertFalse(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, false));
+    }
+
+    @Test
+    void sendEmailIngestionOkInvalidUserInvalidFlowFailed() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(invalidIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(invalidIngestionFlowFileDTO.getOrg().getIpaCode(), invalidUserInfoDTO.getMappedExternalUserId())).thenReturn(invalidUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
```

Comment:
Instead of any MailTo, could we set an expectedMailTo?
This way we will test its right building logic